app/lib/server/functions/saveUser.js

```diff
@@ -146,8 +147,13 @@ function validateUserEditing(userId, userData) {
 
 	const canEditOtherUserInfo = hasPermission(userId, 'edit-other-user-info');
 	const canEditOtherUserPassword = hasPermission(userId, 'edit-other-user-password');
+	const user = Users.findOneById(userData._id);
+	const editingUserRoles = userData.roles
+							&& user.roles.length === userData.roles.length
+							&& user.roles.every((role, index) => role === user[index]);
```

Comment:
I don't think this is necessary, and if you're editing user roles wouldn't `user.roles.length` be different from `userData.roles.length` ?

Reply:
I agree this wouldn't be necessary if the `roles` field was sent only when it's being edited, but the client code sends this field (and many others) in `userData` even when they haven't been updated (that's why I decided to add these checks before throwing the "not allowed to edit" errors, otherwise we'd throw errors even when a user wasn't updating a blocked field).

Reply:
you're correct, that is necessary, but in this case if you add or remove roles the `user.roles.length` will be different from `userData.roles.length`, right? If that's true I guess the best approach is  just create another anon func to compare both arrays elements

Reply:
Ohh, you're right. I've just updated `isEditingUserRoles` to correct its value in this case. I'm not considering changes in the order of the `roles` array's elements as an update.